src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -91,4 +92,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DefineConstants>$(DefineConstants);$(_FrameworkIdentifierForImplicitDefine)$(_FrameworkVersionForImplicitDefine)</DefineConstants>
   </PropertyGroup>
 
+  <!-- Handle XML documentation file settings -->
+  <PropertyGroup Condition="'$(GenerateDocumentationFile)' == 'true' and '$(DocumentationFile)' == ''">
+    <DocumentationFile>$(IntermediateOutputPath)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(GenerateDocumentationFile)' == 'false' and '$(DocumentationFile)' != ''">
+    <_InvalidDocumentationFileSettingsError>GenerateDocumentationFile is false but DocumentationFile is '$(DocumentationFile)', which is an invalid combination.  Either set GenerateDocumentationFile to true (or leave it unset), or leave DocumentationFile unset.</_InvalidDocumentationFileSettingsError>
+  </PropertyGroup>
+  <Target Name="_CheckForInvalidDocumentationFileSettings" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
+    <Error Condition="'$(_InvalidDocumentationFileSettingsError)' != ''" Text="$(_InvalidDocumentationFileSettingsError)" />
+  </Target>
```

Comment:
I'm not sure I get the validation here. Do we really need It? It is very confusing to treat empty, false, and anything but true differently in different places.